### ISSUE TITLE
out of bounds stack buffer read

### ISSUE THREAD

**Issue by 251** [mm2/Little-CMS#20]
Type_NamedColor_Read() creates a "Root" buffer of size 33 without 0-termination (local buffer is uninitalised) and reads 32 bytes from io, also without check for null termination:

``` C
        char Root[33];

        memset(Colorant, 0, sizeof(Colorant));
        if (io -> Read(io, Root, 32, 1) != 1) return NULL;
        if (!_cmsReadUInt16Array(io, 3, PCS)) goto Error;
        if (!_cmsReadUInt16Array(io, nDeviceCoords, Colorant)) goto Error;

        if (!cmsAppendNamedColor(v, Root, PCS, Colorant)) goto Error;
```

cmsAppendNamedColor then applies strncpy with huge cmsMAX_PATH:

``` C
    if (Name != NULL) {

        strncpy(NamedColorList ->List[NamedColorList ->nColors].Name, Name, cmsMAX_PATH-1);
        NamedColorList ->List[NamedColorList ->nColors].Name[cmsMAX_PATH-1] = 0;
```

and strncpy() reads behind Root/Name. You may reproduce the bug as follows:

``` sh
>./configure CC="clang" CFLAGS="-g -O0 -fsanitize=address -fno-omit-frame-pointer -fPIC"
>make
>./utils/psicc/psicc -imin.icc
==24151== ERROR: AddressSanitizer: stack-buffer-overflow on address 0x7fffc1104c24 at pc 0x4130c5 bp 0x7fffc1104270 sp 0x7fffc1103a28
READ of size 1 at 0x7fffc1104c24 thread T0
...
```

base64 representation of min.icc is

```
AAABCgAAAAAAAAAAbm1jbAAAAAAAAAAAAAAAAAAAAAAAAAAAYWNzcAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAABbmNsMgAAAJAAAAB6bmNsMgAAAAAAAAAAAAAAAQAAAABhYWFhYWFh
YWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFh
YWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWEAAAAAAAA=
```

**Comment by mm2** [comment]
Already fixed in 2.10

**Comment by 251** [comment]
"Already" :smile:
